### PR TITLE
TRUNK-5816: Migrate api package tests to Junit 5

### PR DESCRIPTION
**Pull request by achilep**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/hl7/HL7ServiceTest.java [hunk 1]
```diff
@@ -88,8 +90,11 @@ public void processHL7InQueue_shouldCreateHL7InArchiveAfterSuccessfulParsing() t
 		File tempDir = new File(System.getProperty("java.io.tmpdir"), HL7Constants.HL7_ARCHIVE_DIRECTORY_NAME);
 		
 		if (tempDir.exists() && tempDir.isDirectory())
+
 			assertTrue(deleteDirectory(tempDir));
 		
+			assertEquals(true, OpenmrsUtil.deleteDirectory(tempDir));
+
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/hl7/HL7ServiceTest.java:97]:
```suggestion		
```

File: api/src/test/java/org/openmrs/hl7/HL7ServiceTest.java [hunk 2]
```diff
@@ -88,8 +90,11 @@ public void processHL7InQueue_shouldCreateHL7InArchiveAfterSuccessfulParsing() t
 		File tempDir = new File(System.getProperty("java.io.tmpdir"), HL7Constants.HL7_ARCHIVE_DIRECTORY_NAME);
 		
 		if (tempDir.exists() && tempDir.isDirectory())
+
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/hl7/HL7ServiceTest.java:93]:
```suggestion
```

**Comment by teleivo** [comment]
@achilep you will need to resolve these conflicts. after that its ready for me to be merged :)

File: api/src/test/java/org/openmrs/hl7/HL7ServiceTest.java [hunk 3]
```diff
@@ -90,6 +92,7 @@ public void processHL7InQueue_shouldCreateHL7InArchiveAfterSuccessfulParsing() t
 		if (tempDir.exists() && tempDir.isDirectory())
 			assertTrue(deleteDirectory(tempDir));
 		
+			assertEquals(true, OpenmrsUtil.deleteDirectory(tempDir));
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/hl7/HL7ServiceTest.java:95]: this line needs to be removed and the import of `OpenmrsUtil`

Comment (achilep) [reply api/src/test/java/org/openmrs/hl7/HL7ServiceTest.java]: removed but this OpenmrsUtil import is used in several places

**Pull request merged**